tests/collector/test_host_inventory.py

```diff
@@ -13,48 +13,94 @@ class TestRetrieveHosts:
     def test_get_single_pages(self, mocker):
         """When results are in one page."""
         page1 = mocker.MagicMock()
-        page1.json.return_value = \
-            dict(page=1, total=4, per_page=5, results=[0, 1, 2, 3])
+        page1.json.return_value = dict(
+            page=1,
+            total=3,
+            per_page=5,
+            results=[{'id': '0'}, {'id': '1'}, {'id': '2'}, {'id': '3'}],
+        )
+        profiles = mocker.MagicMock()
+        profiles.json.return_value = dict(results=[1, 2, 3])
         retryable = mocker.patch.object(
-            collector.utils, 'retryable', side_effect=[page1]
+            collector.utils, 'retryable', side_effect=[page1, profiles]
         )
 
         headers = {"x-rh-identity": 'identity_b64'}
         results = collector.host_inventory._retrieve_hosts(headers)
 
-        assert results['results'] == list(range(4))
-        retryable.assert_called_once_with(
-            'get',
-            'http://inventory:8080/api/inventory/vX/hosts?page=1',
-            headers={"x-rh-identity": 'identity_b64'}
-        )
+        assert results['results'] == [1, 2, 3]
+        calls = [
+            mocker.call(
+                'get',
+                'http://inventory:8080/api/inventory/vX/hosts?page=1',
+                headers={'x-rh-identity': 'identity_b64'},
+            ),
+            mocker.call(
+                'get',
+                'http://inventory:8080/api/inventory/vX/hosts/'
+                '0,1,2,3/system_profile',
```

Comment:
This is an interesting way to bulk query id's!
It looks like only Host Inv API supports it